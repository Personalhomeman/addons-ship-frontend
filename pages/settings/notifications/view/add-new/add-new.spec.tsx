jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(input => [input, jest.fn()])
}));

import React, { useState } from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Input, Button } from '@bitrise/bitkit';

import AddNew, { Props } from '.';

describe('NotificationSettings View AddNew', () => {
  const defaultProps: Props = {
    onAddEmail: jest.fn()
  };

  it('renders correctly', () => {
    const tree = shallowToJson(shallow(<AddNew {...defaultProps} />));

    expect(tree).toMatchSnapshot();
  });

  it('updates the email value', () => {
    const setEmail = jest.fn();
    (useState as jest.Mock).mockReturnValueOnce(['', setEmail]);

    const email = 'new@email.value';

    const wrapper = shallow(<AddNew {...defaultProps} />);

    wrapper.find(Input).simulate('change', { target: { value: email } });

    expect(setEmail).toHaveBeenCalledWith(email);
  });

  it('calls onAddEmail', () => {
    const email = 'new@email.value';
    (useState as jest.Mock).mockReturnValueOnce([email, jest.fn()]);
    const onAddEmail = jest.fn();

    const wrapper = shallow(<AddNew onAddEmail={onAddEmail} />);
    wrapper.find(Button).simulate('click');

    expect(onAddEmail).toHaveBeenCalledWith(email);
  });
});
