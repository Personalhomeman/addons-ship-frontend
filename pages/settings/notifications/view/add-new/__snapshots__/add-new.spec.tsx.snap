// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationSettings View AddNew renders correctly 1`] = `
<form
  className="container"
  onSubmit={[Function]}
>
  <InputLabel>
    Add New
  </InputLabel>
  <InputContainer>
    <InputContent>
      <Input
        onChange={[Function]}
        placeholder="Email address"
        type="email"
        value=""
      />
    </InputContent>
    <InputContent
      outside={true}
    >
      <Button
        disabled={true}
        level="secondary"
        size="medium"
      >
        Add
      </Button>
    </InputContent>
  </InputContainer>
</form>
`;
